File: CSHARP/diagonal difference.cs
using System;
using System.Collections.Generic;

class Solution
{
    public static int DiagonalDifference(List<List<int>> arr)
    {
        int n = arr.Count;
        int leftToRight = 0;
        int rightToLeft = 0;

        for (int i = 0; i < n; i++)
        {
            leftToRight += arr[i][i];
            rightToLeft += arr[i][n - 1 - i];
        }

        return Math.Abs(leftToRight - rightToLeft);
    }

    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine());
        var matrix = new List<List<int>>();

        for (int i = 0; i < n; i++)
        {
            var row = new List<int>();
            string[] inputs = Console.ReadLine().Split(' ');

            foreach (string input in inputs)
            {
                row.Add(Convert.ToInt32(input));
            }

            matrix.Add(row);
        }

        int result = DiagonalDifference(matrix);
        Console.WriteLine(result);
    }
}

File: PYTHON/diagonal difference.py
def diagonalDifference(arr):
    n = len(arr)
    left_to_right = 0
    right_to_left = 0

    for i in range(n):
        left_to_right += arr[i][i]
        right_to_left += arr[i][n - 1 - i]

    return abs(left_to_right - right_to_left)

# Read input
n = int(input())
arr = []

for _ in range(n):
    row = list(map(int, input().rstrip().split()))
    arr.append(row)

result = diagonalDifference(arr)
print(result)
